pkg/client/client.go

```diff
@@ -62,10 +64,12 @@ const (
 	kubeCloudConfigName = "kube-cloud-config"
 	// cloudCABundleKey is the key in the kube cloud config ConfigMap where the custom CA bundle is located
 	cloudCABundleKey = "ca-bundle.pem"
+	// awsRegionsCacheExpirationDuration is the duration for which the AWS regions cache is valid
+	awsRegionsCacheExpirationDuration = time.Minute * 30
```

Comment:
How often regions list might change? Maybe populate this cache once when controller started?

Reply:
this will simplify things i believe

Reply:
As users have the option to enable/disable some regions at the account level, this information could change while the controller is running. I think it's best to have a cache timeout so that users don't have to kick the pod if their AWS admin enables a new region

Reply:
> As users have the option to enable/disable some regions at the account level
Ok, thanks for explanation!